### PR TITLE
crazy version

### PR DESCRIPTION
**Pull request by bombarder**

### CODE REVIEW

File: src/main/java/Addition.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+public class Addition {
+    public static int add(int a, int b){
```
Comment (sim4ik) [inline src/main/java/Addition.java]: Remove static and consider using of Strategy design pattern.

File: src/main/java/Calculator.java [hunk 2]
```diff
@@ -1,9 +1,37 @@
...
+import java.util.Scanner;
+
 public class Calculator {
     public static void main(String[] args) {
-        System.out.println("hi!");
-        System.out.println("develop");
+
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("Enter digits, please");
+        boolean loop = true;
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: Useless variable. Remove.

File: src/main/java/Calculator.java [hunk 3]
```diff
@@ -1,9 +1,37 @@
...
+        List<String> list = new ArrayList<String>( );
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: Just ArrayList<>();

File: src/main/java/Calculator.java [hunk 4]
```diff
@@ -1,9 +1,37 @@
...
+        while (loop){
+            String str = scanner.nextLine();
+            if (str.equals("=")){
+                break;
+            } else {
+                list.add(str);
+            }
+        }
+
+        int a = Integer.parseInt(String.valueOf(list.get(0)));
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: Really crazy! Simplify it.

File: src/main/java/Calculator.java [hunk 5]
```diff
@@ -1,9 +1,37 @@
...
+        int b = Integer.parseInt(String.valueOf(list.get(2)));
+        String operation = list.get(1);
+
+        switch (operation.charAt(0)) {
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: You can compare strings. No need in char.

File: src/main/java/Calculator.java [hunk 6]
```diff
@@ -1,9 +1,37 @@
...
+            case '+': System.out.println(Addition.add(a,b));
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: Replace future string with constants values.

File: src/main/java/Calculator.java (hunk 2 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: done!

File: src/main/java/Addition.java (hunk 1 above)
Comment (bombarder) [inline src/main/java/Addition.java]: Static was successfully removed.

File: src/main/java/Calculator.java (hunk 3 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: done!

File: src/main/java/Calculator.java (hunk 4 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: done!

File: src/main/java/Calculator.java (hunk 5 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: done.

File: src/main/java/Calculator.java (hunk 6 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: not sure about what should be done

File: src/main/java/Calculator.java [hunk 7]
```diff
@@ -1,9 +1,44 @@
...
+        int b = Integer.parseInt(list.get(2));
+        String operation = list.get(1);
+
+        Dividation dividation = new Dividation();
+        Addition addition = new Addition();
+        Subtraction subtraction = new Subtraction();
+        Multiplication multiplication = new Multiplication();
+
+        switch (operation) {
+            case "+":
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: Make constants of these strings

File: src/main/java/Calculator.java [hunk 8]
```diff
@@ -1,9 +1,44 @@
...
+            } else {
+                list.add(str);
+            }
+        }
+
+        int a = Integer.parseInt(list.get(0));
+        int b = Integer.parseInt(list.get(2));
+        String operation = list.get(1);
+
+        Dividation dividation = new Dividation();
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: Use interface.

File: src/main/java/Calculator.java [hunk 9]
```diff
@@ -1,9 +1,44 @@
...
+        Addition addition = new Addition();
+        Subtraction subtraction = new Subtraction();
+        Multiplication multiplication = new Multiplication();
+
+        switch (operation) {
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: Use Factory method design pattern.

File: src/main/java/Strategy.java [hunk 10]
```diff
@@ -0,0 +1,4 @@
+
+public interface Strategy {
```
Comment (sim4ik) [inline src/main/java/Strategy.java]: Find appropriate name for interface.

File: src/main/java/Calculator.java [hunk 11]
```diff
@@ -1,9 +1,44 @@
...
+        while (true) {
+            String str = scanner.nextLine();
+            if (str.equals("=")) {
+                break;
+            } else {
+                list.add(str);
+            }
+        }
+
+        int a = Integer.parseInt(list.get(0));
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: Add validation in case I enter something wrong.

File: src/main/java/Calculator.java [hunk 12]
```diff
@@ -1,9 +1,44 @@
...
+        int b = Integer.parseInt(list.get(2));
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: Maybe it's better to make it as double instead of int?

File: src/main/java/Calculator.java (hunk 7 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: done

File: src/main/java/Strategy.java (hunk 10 above)
Comment (bombarder) [inline src/main/java/Strategy.java]: done

File: src/main/java/Calculator.java (hunk 12 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: done

File: src/main/java/Calculator.java (hunk 11 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: done

File: src/main/java/Calculator.java (hunk 9 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: done

File: src/main/java/Calculator.java [hunk 13]
```diff
@@ -1,9 +1,55 @@
...
+        while (!scanner.hasNextInt()) {
+            scanner.next();
+            System.out.print("Invalid input please enter a number: ");
+        }
+        secondDigit = scanner.nextDouble();
+        scanner.close();
+
+        CalculationFactory calculationFactory = new CalculationFactory();
+
+        switch (operator) {
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: No need in this switch! You have already had factory.

File: src/main/java/Calculator.java [hunk 14]
```diff
@@ -1,9 +1,55 @@
...
+
+        System.out.println("Enter second digit, please");
+        while (!scanner.hasNextInt()) {
+            scanner.next();
+            System.out.print("Invalid input please enter a number: ");
+        }
+        secondDigit = scanner.nextDouble();
+        scanner.close();
+
+        CalculationFactory calculationFactory = new CalculationFactory();
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: No need to create it through "new" always. Use Singleton design pattern.

File: src/main/java/Calculator.java [hunk 15]
```diff
@@ -1,9 +1,55 @@
...
+        System.out.println("Enter first digit, please");
+        while (!scanner.hasNextInt()) {
+            scanner.next();
+            System.out.println("Invalid input please enter a number: ");
+        }
+        firstDigit = scanner.nextDouble();
+
+        System.out.println("Enter operator (+, -, *, /): ");
+        operator = scanner.next();
+        while (!operator.equals("+") && !operator.equals("-") && !operator.equals("*") && !operator.equals("/")) {
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: Plese move validation to separate class.

Comment (sim4ik) [inline src/main/java/Calculator.java]: You can create another method for you validation logic.

File: src/main/java/Calculator.java [hunk 16]
```diff
@@ -1,9 +1,55 @@
...
-        System.out.println("hi!");
-        System.out.println("develop");
+
+        Scanner scanner = new Scanner(System.in);
+        double firstDigit;
+        double secondDigit;
+        String operator;
+
+        System.out.println("Enter first digit, please");
+        while (!scanner.hasNextInt()) {
```
Comment (sim4ik) [inline src/main/java/Calculator.java]: Please, try not to dublicate your code.

File: src/main/java/Calculator.java (hunk 13 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: done

File: src/main/java/Calculator.java (hunk 14 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: done

File: src/main/java/Calculator.java (hunk 16 above)
Comment (bombarder) [inline src/main/java/Calculator.java]: done

**Pull request merged**